sql/hive/src/test/scala/org/apache/spark/sql/hive/test/TestHiveSingleton.scala

```diff
@@ -27,9 +27,9 @@ import org.apache.spark.sql.hive.client.HiveClient
 
 trait TestHiveSingleton extends SparkFunSuite with BeforeAndAfterAll {
   override protected val enableAutoThreadAudit = false
-  protected val spark: SparkSession = TestHive.sparkSession
-  protected val hiveContext: TestHiveContext = TestHive
-  protected val hiveClient: HiveClient =
+  protected lazy val spark: SparkSession = TestHive.sparkSession
+  protected lazy val hiveContext: TestHiveContext = TestHive
+  protected lazy val hiveClient: HiveClient =
```

Comment:
Could you add some comment to explain why these `lazy` are required, please?

Reply:
> Could you add some comment to explain why these three lazy are required, please?

OK, I want to make the spark.sql.test.master configuration effective in the test case.